standard/toolchain-android32.cmake

```diff
@@ -1,3 +1,5 @@
 set(ARM_TARGET "armeabi-v7a")
 set(ANDROID_NDK_TOOL_PREFIX arm-linux-androideabi)
+set(CMAKE_SYSTEM_PROCESSOR arm)
+set(CMAKE_SIZEOF_VOID_P 4)
```

Comment:
afaik CMake normally sets `CMAKE_SIZEOF_VOID_P` via a test-compile. We can force its value and have been doing so, but since the goal here is to do things in more of the standard way is there a way we can let CMake figure it out?